hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ConfigCheck.java

```diff
@@ -179,7 +171,8 @@ public int getId() {
     @Override
     public void writeData(ObjectDataOutput out) throws IOException {
         out.writeUTF(groupName);
-        out.writeUTF(groupPassword);
+        //TODO @tkountis remove in 4.0
+        out.writeUTF(EMPTY_PWD);
```

Comment:
I included some short explanation in the TDD the other day, but its worth mentioning here too.
The JoinRequest is happening _before_ a member is part of a cluster, therefore the getClusterVersion() always returns the node's codebase. 

Having that in mind, the above snippet wouldn't make sense, because it will be part of a 3.9 release, meaning that the check will always be false. Therefore, sending empty-pwd by default, and if the receiving node is >=3.8.2 it will know to ignore whatever value is in this field.